sources/core/Stride.Core/Utilities.cs

```diff
@@ -717,88 +611,34 @@ public static bool Compare<T>(ICollection<T> left, ICollection<T> right)
         /// <param name="right">The colllection to compare to.</param>
         /// <returns>True if lists are sequentially equal. False otherwise.</returns>
         /// <remarks>Concrete List is favored over interface to avoid enumerator object allocation.</remarks>
+        [Obsolete("Use StrideCoreExtensions.SequenceEqualAllowNull")]
         public static bool Compare<T>(List<T> left, List<T> right)
-        {
-            if (ReferenceEquals(left, right))
-                return true;
-            if (ReferenceEquals(left, null) || ReferenceEquals(right, null))
-                return false;
-
-            if (left.Count != right.Count)
-                return false;
-
-            var comparer = EqualityComparer<T>.Default;
-            for (int i = 0; i < left.Count; i++)
-            {
-                if (!comparer.Equals(left[i], right[i]))
-                    return false;
-            }
-
-            return true;
-        }
+            => left.SequenceEqualAllowNull(right, null);
 
         /// <summary>
         /// Swaps the value between two references.
         /// </summary>
         /// <typeparam name="T">Type of a data to swap.</typeparam>
         /// <param name="left">The left value.</param>
         /// <param name="right">The right value.</param>
-        public static void Swap<T>(ref T left, ref T right)
-        {
-            var temp = left;
-            left = right;
-            right = temp;
-        }
+        public static void Swap<T>(ref T left, ref T right) => (right, left) = (left, right);
 
-        /// <summary>
-        /// Suspends current thread for a <see cref="sleepTime"/>.
-        /// </summary>
-        /// <param name="sleepTime">The duration of sleep.</param>
-        public static void Sleep(TimeSpan sleepTime)
+        /// <summary>Suspends the current thread for a duration.</summary>
+        /// <param name="duration">The duration of sleep. Must be non-negative.</param>
+        public static void Sleep(TimeSpan duration)
         {
-            var ms = (long)sleepTime.TotalMilliseconds;
-            if (ms < 0 || ms > int.MaxValue)
-            {
-                throw new ArgumentOutOfRangeException(nameof(sleepTime), "Sleep time must be a duration less than '2^31 - 1' milliseconds.");
-            }
-            NativeInvoke.Sleep((int)ms);
+            if (duration.Ticks < 0)
+                throw new ArgumentOutOfRangeException(nameof(duration));
+            Thread.Sleep(duration);
```

Comment:
Is Thread.Sleep behaving the same as the native one - I think it was calling a C function?

Reply:
I see [it](https://github.com/sinkingsugar/NativePath/blob/master/source/NativeThreading.cpp) uses `usleep` or `std::this_thread::sleep_for(std::chrono::milliseconds(milliseconds));`.

`System.Threading.Thread.Sleep` uses `Kernel32!Sleep` [here](https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.CoreLib/src/System/Threading/Thread.Windows.cs#L19) or on Unix its much more [involved](https://github.com/dotnet/runtime/blob/9d6396deb02161f5ee47af72ccac52c2e1bae458/src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.ThreadWaitInfo.Unix.cs#L391) but it'll be compatbile.

The biggest difference imho is that .NET accepts `Timeout.Infinite` which is a negative value, so that's why I didn't obsolete `Utilities.Sleep` although most likely that check is not practically necessary.